### PR TITLE
Updates for Flutter 3.0.0

### PR DESCRIPTION
**Pull request by X-SLAYER**

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Looks great! Looking forward to merging

File: lib/src/qr_code_scanner.dart [hunk 1]
```diff
@@ -65,7 +65,7 @@ class _QRViewState extends State<QRView> {
   void initState() {
     super.initState();
     _observer = LifecycleEventHandler(resumeCallBack: updateDimensions);
-    WidgetsBinding.instance!.addObserver(_observer);
+    WidgetsBinding.instance.addObserver(_observer);
```
Comment (aaassseee) [inline lib/src/qr_code_scanner.dart:68]: Fatal errors can be caused non-nullable widget binding instance due to lower dart version
I think you should also change the dart version range from
https://github.com/juliuscanute/qr_code_scanner/blob/f55ba22b33ab9d0d88c71968d3ba6eaed8f048eb/pubspec.yaml#L8
to
``` yaml
  sdk: '>=2.17.0 <3.0.0'
```

**Review by aaassseee — CHANGES_REQUESTED** [review]

File: lib/src/qr_code_scanner.dart (hunk 1 above)
Comment (jack24254029) [reply lib/src/qr_code_scanner.dart:68]: change the flutter version too.

```flutter: ">=3.0.0"```

**Comment by shivam-modi** [comment]
When will this Pull Request is going to be merged? It is breaking the flow of my app which I recently upgraded to flutter 3.0

**Comment by X-SLAYER** [comment]
> When will this Pull Request is going to be merged? It is breaking the flow of my app which I recently upgraded to flutter 3.0

you can use it until the PR gets merged 

```yaml
qr_code_scanner: 
  git:
    url: https://github.com/X-SLAYER/qr_code_scanner.git
    ref: flutter-beta
```

**Comment by shivam-modi** [comment]
> > When will this Pull Request is going to be merged? It is breaking the flow of my app which I recently upgraded to flutter 3.0
> 
> you can use it until the PR gets merged
> 
> ```yaml
> qr_code_scanner: 
>   git:
>     url: https://github.com/X-SLAYER/qr_code_scanner.git
>     ref: flutter-beta
> ```

I had already done this, but it should be merged as soon as possible.

**Comment by X-SLAYER** [comment]
yes for sure but the author seems inactive i think

**Comment by gabaudette** [comment]
This need to be merge ASAP

**Review by Abdullahjavaid0331 — APPROVED** [review]

**Review by AugustoResende — APPROVED** [review]

File: lib/src/qr_code_scanner.dart (hunk 1 above)
Comment (aaassseee) [reply lib/src/qr_code_scanner.dart:68]: > change the flutter version too.
> 
> `flutter: ">=3.0.0"`

@X-SLAYER Is this changed with this pull request? I didn't find any changes on yaml file. Moreover, ignore my suggestion on sdk version change. Changing Flutter version is more accurate.

**Review by bilalsammour — APPROVED** [review]

**Comment by RaulUrtecho** [comment]
Yes one approval more I need this asap

**Comment by serhatuysal** [comment]
please!!

**Comment by bilalsammour** [comment]
Waiting as well.

**Comment by kuromukira** [comment]
The waiting game.

**Comment by Zersya** [comment]
> yes for sure but the author seems inactive i think

he's on vacation till 23 may, I think 
https://github.com/juliansteenbakker/mobile_scanner/pull/123#issuecomment-1125949082

**Comment by jingjiewei** [comment]
```
qr_code_scanner: 
  git:
    url: https://github.com/X-SLAYER/qr_code_scanner.git
    ref: flutter-beta
```
This version cause an black screen and can not scan on android, anyone else has the same issue?

**Comment by allyssonmast** [comment]
> black screen

Same issue here. This version works on first access (after granting permission to use camera). On a second try it shows a black screen and it only works again if you minimize the app and open it again (but it's not the right way to work)

**Comment by allyssonmast** [comment]
> ```
> qr_code_scanner: 
>   git:
>     url: https://github.com/X-SLAYER/qr_code_scanner.git
>     ref: flutter-beta
> ```
> 
> This version cause an black screen and can not scan on android, anyone else has the same issue?

Same issue here. This version works on first access (after granting permission to use camera). On a second try it shows a black screen and it only works again if you minimize the app and open it again (but it's not the right way to work)

**Comment by ricardo-kowalski** [comment]
> 



> > ```
> > qr_code_scanner: 
> >   git:
> >     url: https://github.com/X-SLAYER/qr_code_scanner.git
> >     ref: flutter-beta
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This version cause an black screen and can not scan on android, anyone else has the same issue?
> 
> Same issue here. This version works on first access (after granting permission to use camera). On a second try it shows a black screen and it only works again if you minimize the app and open it again (but it's not the right way to work)

Me too. The problem is the same presented in issue #277

**Comment by shivam-modi** [comment]
Regarding the 

> ```
> qr_code_scanner: 
>   git:
>     url: https://github.com/X-SLAYER/qr_code_scanner.git
>     ref: flutter-beta
> ```
> 
> This version cause an black screen and can not scan on android, anyone else has the same issue?

For me the black screen appears at first launch. I found a workaround for this to issue used QrViewController instance to resume camera. 

inside initState
`controller.resumeCamera();`

**Comment by nimi0112** [comment]
> > ```
> > qr_code_scanner: 
> >   git:
> >     url: https://github.com/X-SLAYER/qr_code_scanner.git
> >     ref: flutter-beta
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This version cause an black screen and can not scan on android, anyone else has the same issue?
> 
> 
> Same issue here. This version works on first access (after granting permission to use camera). On a second try it shows a black screen and it only works again if you minimize the app and open it again (but it's not the right way to work)

same for me

**Comment by jingjiewei** [comment]
> > > ```
> > > qr_code_scanner: 
> > >   git:
> > >     url: https://github.com/X-SLAYER/qr_code_scanner.git
> > >     ref: flutter-beta
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This version cause an black screen and can not scan on android, anyone else has the same issue?
> > 
> > 
> > Same issue here. This version works on first access (after granting permission to use camera). On a second try it shows a black screen and it only works again if you minimize the app and open it again (but it's not the right way to work)
> 
> same for me

check this workround https://github.com/juliuscanute/qr_code_scanner/issues/538#issuecomment-1133883828

**Comment by xshiwei** [comment]
尽快合并

**Comment by RaulUrtecho** [comment]
I can't believe this has been stuck for more than 2 weeks.

**Comment by AugustoResende** [comment]
@juliansteenbakker

**Comment by juliansteenbakker** [comment]
Very sorry for the delay (lots of packages to maintain). So i've merged another pr which causes some issues with this one, but  i will cherry pick the changes and update the package. Thanks for the contribution!

**Pull request closed without merging**